File: iosApp/iosApp/Helpers/OAuthCallbackHandler.swift
//
//  OAuthCallbackHandler.swift
//  iosApp
//
//  Singleton handler for OAuth callbacks
//

import Foundation
import Combine

/// Singleton handler for OAuth callbacks.
/// iOSApp posts callback URLs here, and AuthViewModelWrapper collects them.
class OAuthCallbackHandler: ObservableObject {
    static let shared = OAuthCallbackHandler()

    @Published var callbackUrl: URL?

    private init() {}

    func handleCallback(_ url: URL) {
        // Verify this is an OAuth callback
        guard url.scheme == "kluvs",
              url.host == "auth",
              url.path == "/callback" else {
            return
        }

        print("OAuthCallbackHandler: Received callback: \(url)")
        callbackUrl = url
    }

    func clearCallback() {
        callbackUrl = nil
    }
}

File: composeApp/src/androidMain/kotlin/com/ivangarzab/kluvs/ui/OAuthCallbackHandler.kt
package com.ivangarzab.kluvs.ui

import kotlinx.coroutines.flow.MutableSharedFlow
import kotlinx.coroutines.flow.SharedFlow
import kotlinx.coroutines.flow.asSharedFlow

/**
 * Singleton handler for OAuth callbacks.
 *
 * MainActivity posts callback URLs here, and the AuthViewModel collects them.
 */
object OAuthCallbackHandler {
    private val _callbacks = MutableSharedFlow<String>(replay = 0, extraBufferCapacity = 1)
    val callbacks: SharedFlow<String> = _callbacks.asSharedFlow()

    fun handleCallback(url: String) {
        _callbacks.tryEmit(url)
    }
}
